### ISSUE TITLE
list merge strategy append_unique does not work for lists of dicts

### ISSUE THREAD

**Issue by olifur** [toumorokoshi/deepmerge#20]
Hi developers, especially @morph027 ,

I get an error when trying to apply list merge strategy `append_unique` for lists of dictionaries.
I am using deepmerge 1.0.1 and python 3.7.7.

When I am running the following code
```python
from deepmerge import Merger

my_merger = Merger(
    # pass in a list of tuple, with the
    # strategies you are looking to apply
    # to each type.
    [
        (list, ["append_unique"]),
        (dict, ["merge"]),
        (set, ["union"])
    ],
    # next, choose the fallback strategies,
    # applied to all other types:
    ["override"],
    # finally, choose the strategies in
    # the case where the types conflict:
    ["override"]
)
base = {"foo": ["bar"]}
next = {"foo": ["bar","baz"]}

result = my_merger.merge(base, next)

assert result == {'foo': ['bar', 'baz']}

base = {"foo": [{"bar": ["bob"]}]}
next = {"foo": [{"bar": ["baz"]}]}

result = my_merger.merge(base, next)

assert result == {'foo': [{'bar', ["bob"]}, {"bar": ["baz"]}]}

```

I get the following exception

```bash
python3 test_merge.py 
Traceback (most recent call last):
  File "test_merge.py", line 29, in <module>
    result = my_merger.merge(base, next)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/merger.py", line 33, in merge
    return self.value_strategy([], base, nxt)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/merger.py", line 43, in value_strategy
    return strategy(self, path, base, nxt)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/strategy/core.py", line 35, in __call__
    ret_val = s(*args, **kwargs)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/strategy/dict.py", line 23, in strategy_merge
    base[k] = config.value_strategy(path + [k], base[k], v)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/merger.py", line 43, in value_strategy
    return strategy(self, path, base, nxt)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/strategy/core.py", line 35, in __call__
    ret_val = s(*args, **kwargs)
  File "/home/horst/venv/lib64/python3.7/site-packages/deepmerge/strategy/list.py", line 29, in strategy_append_unique
    base_as_set = set(base)
TypeError: unhashable type: 'dict'
```

Best,
Oliver

**Comment by morph027** [comment]
Hm, i see. I was the one who added the PR, but the code itself is from https://github.com/toumorokoshi/deepmerge/pull/7

Right now, i do not have a clue how to solve this easily. Might require some recursion...

**Comment by toumorokoshi** [comment]
Thanks for the bug report!

So the gist of it is that uniqueness cannot be determined for unhashable objects. 

I have a fix with #20, if someone wants to review. If there's no qualms with the approach I'll get that merged in.
